__tests__/app.test.js

```diff
@@ -273,4 +273,57 @@ describe('app.js', () => {
             })
         })
     })
+    describe('PATCH /api/articles/:article_id', () => {
+        test('200: patch increases the votes on the requested article', () => {
+            const input = {
+                inc_votes : 1 
+            }
+            return request(app).patch('/api/articles/1').send(input)
+            .expect(200)
+            .then(({body}) => {
+                const {article} = body
+                expect(article.votes).toBe(101)
+            })
+        })
+        test('200: patch decreases the votes on the requested article', () => {
+            const input = {
+                inc_votes : -99
+            }
+            return request(app).patch('/api/articles/1').send(input)
+            .then(({body}) => {
+                const {article} = body
+                expect(article.votes).toBe(1)
+            })
+        })
+        test('400: patch fails if given wrong key', () => {
+            const input = {
+                title : 'new title'
+            }
+            return request(app).patch('/api/articles/1').send(input)
+            .expect(400)
+            .then(({body}) => {
+                expect(body.msg).toBe('Bad request')
+            })
+        })
+        test('400: patch fails if given wrong data', () => {
+            const input = {
+                inc_votes : 'test'
+            }
+            return request(app).patch('/api/articles/1').send(input)
+            .expect(400)
+            .then(({body}) => {
+                expect(body.msg).toBe('Bad request')
+            })
+        })
+        test('400: patch fails if given id that does not refer to an article', () => {
+            const input = {
+                inc_votes : 1 
+            }
+            return request(app).patch('/api/articles/20000').send(input)
+            .expect(400)
+            .then(({body}) => {
+                expect(body.msg).toBe('Bad request')
+            })
+        })
```

Comment:
(Optional) Status 200, if inc_votes is missing, should not update article and return original single article object